### ISSUE TITLE
What model compression operations did yolov8 perform when saving the model

### ISSUE THREAD

**Issue by bug-cat-iu** [ultralytics/ultralytics#1479]
### Search before asking

- [x] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I noticed that if yolov8 suddenly interrupts the training model, the saved model at this time will be much larger than the model saved after natural training. I would like to know what model compression operations he performed when the model was saved after automatic training.

**Comment by glenn-jocher** [comment]
@bug-cat-iu below explanation applies to YOLOv5, but also applies to YOLOv8 since we share a similar training and checkpointing strategy:

👋 Hello! Thanks for asking about **training checkpoints**.  YOLOv5 🚀 checkpoints should be about 4X the size of final trained checkpoints, as they carry not just a FP16 model, but a FP16 [EMA](https://www.tensorflow.org/api_docs/python/tf/train/ExponentialMovingAverage) and an FP32 optimizer of the same size as the model (each model parameter has its own FP32 gradient saved within the optimizer). Final checkpoints contain only an FP16 model, with the EMA and optimizer both stripped after the final epoch of training:
https://github.com/ultralytics/yolov5/blob/b4a29b5a8d63a8c2d4a8929942b44e8969c5dddd/train.py#L423-L425

The `strip_optimizer()` function updates the checkpoint dictionary value for `model`, replacing it with `ema`, and sets `ema` and `optimizer` keys to `None`, which will reduce the checkpoint size by 3/4.
https://github.com/ultralytics/yolov5/blob/b4a29b5a8d63a8c2d4a8929942b44e8969c5dddd/utils/general.py#L741-L755

Final trained checkpoints after `strip_optimizer()` should match the README table sizes below.
<p align="center"><img width="700" alt="YOLOv5 Models" src="https://github.com/ultralytics/yolov5/releases/download/v1.0/model_comparison.png"></p>

You can also run `strip_optimizer()` manually on any checkpoint to convert it into finalized weights ready to train a new model:
```python
from utils.general import strip_optimizer

strip_optimizer('path/to/best.pt')
!python train.py --weights path/to/best.pt  # use best.pt to train a new model
```

### Model Exports

Exported models may be larger or smaller than their PyTorch starting points depending on their quantization. i.e. FP16 models will be the same size, FP32 models will be 2X the size, and INT8 models will be 0.5X the size. See YOLOv5 [Export tutorial](https://github.com/ultralytics/yolov5/issues/251) for quantization options on model export.

Good luck 🍀 and let us know if you have any other questions!

**Comment by legeiger** [comment]
Just a quick addition: to import strip_optimizer into your code using the pip package:
`from ultralytics.utils.torch_utils import strip_optimizer`

**Comment by glenn-jocher** [comment]
@legeiger you're absolutely correct! If you're using YOLOv8 installed as a PyPI package, you would import the `strip_optimizer` function from the `ultralytics.utils.torch_utils` module. This utility is very convenient for preparing a trained model for deployment.

Using `strip_optimizer`, you can remove the optimizer and any additional training-related states from your checkpoint when you want to use the model purely for inference or deployment, thus reducing the file size significantly. This results in a lighter model that's optimized for faster loading and potentially lower memory footprint during inference.

If you face unexpected interruptions during training, you can always manually call `strip_optimizer` to clean up your last saved checkpoint. This would mimic the saving process of a naturally completed training session, where the optimizer states are stripped out after the last epoch.

Keep in mind that once you strip the optimizer from the checkpoint, you cannot resume training from that point without reinitializing a new optimizer state.

I'm glad you're exploring the finer details of our saving mechanisms, and I hope your model training and deployment go smoothly. If you have any more questions about YOLOv8 or other aspects of our work at Ultralytics, feel free to reach out! 🌟
